crates/uv/tests/it/common/mod.rs

```diff
@@ -563,6 +564,21 @@ impl TestContext {
         self
     }
 
+    // Unsets the git credential helper using temp home gitconfig
+    pub fn with_git_credential_helper_blocked(self) -> Self {
+        let path = self.home_dir.join(".gitconfig");
+        let mut file = fs_err::OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&path)
+            .expect("Failed to open gitconfig file");
+
+        writeln!(file, "[credential]\n\thelper =\n")
+            .expect("Failed to write credential helper to gitconfig");
+
+        self
+    }
```

Comment:
We may be able to leverage the existing assert_fs support, e.g.

```suggestion
    pub fn with_unset_git_credential_helper(self) -> Self {
        let git_config = self.home_dir.child(".gitconfig");
        git_config
            .write_str(indoc! {r"
                [credential]
                    helper =
            "})
            .expect("Failed to unset git credential helper");
        self
    }
```